woocommerce/woocommerce-gateway-stripe
# feat: payment method capability status

# Changes proposed in this Pull Request:

Helps with https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1874 - splitting into multiple PRs

- Fixes the styling of the "Pending activation" pill on the onboarding view
- Ensures the "Pending activation" pill is displayed when the status of the payment method capability is `pending`
- Moved the `PaymentMethodCheckbox` component from the `client/components` directory to a location that is closer to where it is used (`client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/index.js`)

**PLEASE NOTE**: in the "old" settings, we showed the "Pending activation" information _also_ when the capability was not present (or `inactive`). I'm not sure that's accurate, so now we're displaying "Pending activation" _only_ when the capability status is `pending`.

Before:
![Screen Shot 2021-10-05 at 3 46 54 PM](https://user-images.githubusercontent.com/273592/136108619-3abdb6cf-da10-43bc-951b-cd0022eb7190.png)

After:
![Screen Shot 2021-10-05 at 4 25 28 PM](https://user-images.githubusercontent.com/273592/136108590-0bac2f53-e549-4c47-9e6f-c1fbdd57c320.png)

After - with fees:
![Screen Shot 2021-10-05 at 4 31 51 PM](https://user-images.githubusercontent.com/273592/136108559-44308e2f-0e16-463c-bf80-b2b507362e28.png)


# Testing instructions
- Ensure you have the `_wcstripe_feature_upe_settings` flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard
- Go to the second step
- The capability status pill now reflects the contents of the `capability` object in your account
- The same is also true for the main settings page, with UPE enabled: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe
